### ISSUE TITLE
Gradient Clipping with SAM optimizer

### ISSUE THREAD

**Issue by stardust-88** [davda54/sam#64]
How to include the gradient clipping line when using the SAM optimizer?
Normally, it's like:
  ```
     loss.backward()
     torch.nn.utils.clip_grad_norm_(model.parameters(), args.max_grad_norm)
     optmizer.step()
     scheduler.step()
```
But how would look like when using SAM optimizer, is it like this?
```
   loss.backward()
   torch.nn.utils.clip_grad_norm_(model.parameters(), args.max_grad_norm)
   optimizer.first_step(zero_grad=True)
   loss_fn(model(inputs),targets).backward()
   torch.nn.utils.clip_grad_norm_(model.parameters(), args.max_grad_norm)
   optimizer.second_step(zero_grad=True)
```

**Comment by stardust-88** [comment]
Please address this.

**Comment by davda54** [comment]
Hi, your implementation makes sense to me. Are there any problems with it not working?

**Comment by stardust-88** [comment]
Hi! Still have to test, and am about to. Theoretically it seems fine to me too because we are doing two backward passes and therefore gradients need to be clipped two times, right?
